chain-vote/src/lib.rs

```diff
@@ -382,9 +460,17 @@ mod tests {
         tally.add(&e2.0, 3);
         tally.add(&e3.0, 4);
 
-        let (_, tds1) = tally.finish(m1.secret_key());
-        let (_, tds2) = tally.finish(m2.secret_key());
-        let (ts, tds3) = tally.finish(m3.secret_key());
+        let (_, tds1) = tally.finish(&mut rng, m1.secret_key());
+        let (_, tds2) = tally.finish(&mut rng, m2.secret_key());
+        let (ts, tds3) = tally.finish(&mut rng, m3.secret_key());
+
+        // check that the verify shares are correct for each participants
+        assert_eq!(verify_decrypt_share(&tally, &m1.public_key(), &tds1), true);
```

Comment:
Is this check redundant now that shares are verified with the tally?

Reply:
yep, just kept one for checking that the function fails with incorrect key